contracts/entry-point/src/contract.rs

```diff
@@ -85,6 +91,48 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> ContractResult<Response> {
     match msg {
+        ExecuteMsg::AxelarSwapAndAction {
+            user_swap,
+            min_coin,
+            timeout_timestamp,
+            post_swap_action,
+            affiliates,
+            recovery_addr,
+        } => {
+            // Store all parameters into a temporary storage.
+            SWAP_AND_ACTION_REQUEST_TEMP_STORAGE.save(
+                deps.storage,
+                &SwapActionTempStorage {
+                    user_swap: user_swap.clone(),
+                    min_coin: min_coin.clone(),
+                    timeout_timestamp,
+                    post_swap_action: post_swap_action.clone(),
+                    affiliates: affiliates.clone(),
+                    funds: info.funds.clone(),
+                    recovery_addr,
+                },
+            )?;
+
+            // Then call ExecuteMsg::SwapAndAction using a SubMsg.
+            let sub_msg = SubMsg::reply_always(
+                CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr: env.contract.address.to_string(),
+                    msg: to_binary(&ExecuteMsg::SwapAndAction {
+                        user_swap: user_swap.clone(),
+                        min_coin: min_coin.clone(),
+                        timeout_timestamp,
+                        post_swap_action: post_swap_action.clone(),
+                        affiliates: affiliates.clone(),
+                    })?,
+                    funds: vec![],
```

Comment:
Within SwapAndAction we expect funds to be sent with the calling message using one_coin, so either we need to pass on the funds here or change the check in SwapAndAction, my pref/suggestion would be just to pass the funds in this call

https://github.com/skip-mev/skip-api-contracts/blob/1df0b3496a1e0bde8b46d3aad64c6c44aa6b067a/contracts/entry-point/src/execute.rs#L45